### DISCUSSION TITLE
How to style an input type range sliders thumb and track

### DISCUSSION THREAD

**Discussion by djkato** [tailwindlabs/tailwindcss#8748] [Help] [answered]
Same as in [#6972](https://github.com/tailwindlabs/tailwindcss/discussions/6972), I don't know how I would go about styling the pseudo elements of a range type input slider, specifically it's thumb and track.
https://toughengineer.github.io/demo/slider-styler/slider-styler.html
http://danielstern.ca/range.css/#/
All slider customization requires access to these, but even putting it into the `<style>` element (Vue) ignores the regular css:
```css
input[type=range]::-webkit-slider-runnable-track,
input[type=range]::-ms-track,
input[type=range]::-moz-range-track{
/*stuff here*/
}
input[type=range]::-moz-range-thumb,
input[type=range]::-ms-thumb,
input[type=range]::-webkit-slider-thumb{
/*stuff here*/
}
```
The tailwind doc doesn't mention anything about input elements in general from what I tried.

**Comment by rohmann** [comment, marked as answer]
Hi @djkato,

You can see all the pseudo classes included natively here: https://tailwindcss.com/docs/hover-focus-and-other-states#pseudo-class-reference. The range input pseudo classes aren't included. I think Tailwind likes to steer clear of non standard cases like this because it's a whole lot of nuance to support over time. For years the range input has been a strange hodge-podge of selectors across broweser. I'm sure it's something they would consider down the road if a specification emerged to make it consistent

I've not tried it in all browsers, but I've noticed that Chrome based browsers are finicky when it comes to multiple selectors. Breaking them out into their own rule sets (like the examples you've shared) seems to help.

Example: https://play.tailwindcss.com/yFQGknjCqn

**Reply by djkato** [reply]
Thanks! https://tailwindcss.com/docs/adding-custom-styles#using-css-and-layer 
Got curious about that @layer and it lead me to that doc link^ Seems its exactly what I need

**Reply by rohmann** [reply]
Ok great! Good luck getting it all dialed in.

**Comment by aayush301** [comment]
According to recent versions of Tailwind CSS, If you want, you can add these styles right inside your HTML file:
class="[&::-webkit-slider-runnable-track]:rounded-full ..."

https://play.tailwindcss.com/9ayu07qkQh

**Reply by guzily** [reply]
it seems not working for "[&::-webkit-slider-thumb]"

**Reply by aayush301** [reply]
Have you checked the link because it works in my case? Maybe you could be using mozilla firefox or edge. If that is the case, then you will have to add respective prefixes.

**Reply by guzily** [reply]
@aayush301  yeah ,I checked the link with Chrome.  However , it didnt work. Are there any mistakes in my codes ? Thanks please!
<img width="1377" alt="image" src="https://user-images.githubusercontent.com/74231693/223989992-a239544c-5d8e-4d56-9b81-cf6ca0f76afd.png">

**Reply by aayush301** [reply]
Since you are also interested in changing the default style of thumb, you will first have to make its appearance-none. Also don't forget to set its height, width, and roundness.

**Reply by guzily** [reply]
@aayush301  Thanks a lot. It did help
`<input type="range" class="appearance-none bg-transparent [&::-webkit-slider-runnable-track]:rounded-full [&::-webkit-slider-runnable-track]:bg-black/25 [&::-webkit-slider-thumb]:appearance-none [&::-webkit-slider-thumb]:h-[50px] [&::-webkit-slider-thumb]:w-[50px] [&::-webkit-slider-thumb]:rounded-full [&::-webkit-slider-thumb]:bg-purple-500" />`

**Reply by aayush301** [reply]
Glad it worked!!

**Reply by apeksha0513** [reply]
hey can you please help me. I need to add color till scroll. Just like range fill but with tailwind only on custom css 
and I would really appreciate if you could reply it asap

**Reply by aayush301** [reply]
@apeksha0513 I think some javascript is required for this. maybe stackoverflow's this answer can help you.
https://stackoverflow.com/a/63941494/16785998

**Reply by zhaodong-wang** [reply]
Seems like it does not work with the newest version? https://play.tailwindcss.com/wjpgEw00Bk

![image](https://github.com/tailwindlabs/tailwindcss/assets/8356125/e62d1f09-e5c3-4f0a-a804-1849949518b1)

**Reply by FredTreg** [reply]
@zhaodong-wang use `class` instead of `className`

**Reply by ghost** [reply]
Thanks :)

**Reply by fuad-hasan-safat** [reply]
how can I change the color of blue part

**Reply by yongvuthivann** [reply]
Is there any suggestion the blue part?

**Reply by stanevidanza** [reply]
This is my solution, it's working for me: `h-2 w-full cursor-pointer appearance-none rounded-md bg-gray-200 [&::-webkit-slider-thumb]:!bg-primary`
